### PR TITLE
TEP 0109: Better structured provenance retrieval in Tekton Chains

### PR DESCRIPTION
**Pull request by ywluogg**
Finalize the discussions and reviews regarding to support structured results and params in Tekton Chains.

The original PR was here: https://github.com/tektoncd/community/pull/697 but the merging issue closed the original thread.

I'm reopening it in this PR.

### CODE REVIEW

**Comment by ywluogg** [comment]
Follow-ups from #697 

> It might help to refocus this TEP on the desired end state, and move the middle ground impl to Implementation Plan and/or Risks and Mitigations.
> 
> We should also look out to fill out the sections for Drawbacks - though I think this should focus on the end goal rather than in-between steps.

I've moved the middle ground implementation into a subsection and added Drawbacks for this one @wlynch

File: teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md
Comment (ywluogg) [reply teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md]: Ah I brought up Trusted resources because I wanted to mention that currently in Tekton Pipelines, only the controller can change the `status` field. I reworded it a bit and shifted my main argument to be that we don't need to predefine a schema in Tasks. PTAL thanks!

File: teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md
Comment (ywluogg) [reply teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md:416]: SG! I created one in Pipelines: https://github.com/tektoncd/pipeline/issues/5455

Comment (wlynch) [reply teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md:416]: I'd link to it in this proposal and mention we'll likely aim to deprecate this once the linked issue is implemented.

**Review by wlynch — COMMENTED** [review]
/lgtm - minor comments but nothing worth blocking on.

holding off on approval for n+1 review (ideally a pipeline maintainer). you may want to sync with @jerop - there was discussion on using the `artifact` naming elsewhere in pipelines, so you may want to sort that out to avoid overlap.

File: teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md
Comment (dibyom) [inline teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md:226]: Do the artifacts have to provide the "type" (i.e. maven, perforce etc.) as part of the name? Or is that just for illustrating these examples?

**Review by dibyom — APPROVED** [review]

**Comment by lbernick** [comment]
/assign @vdemeester 
@jagathprakash PTAL
@wlynch want to double check if you are good leaving an approval now that Dibyo has approved?

File: teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md
Comment (jagathprakash) [inline teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md:237]: Both the algorithm name and the digest would not have ",". Could we have a comma separated list of sha:digest which we could parse out. This is just to be feature complete wrt to the DigestSet in the SLSA specification.

**Comment by wlynch** [comment]
> @wlynch want to double check if you are good leaving an approval now that Dibyo has approved?

I'm surprised prow didn't pick up on the last LGTM - sorry about that!

/lgtm

File: teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md
Comment (ywluogg) [reply teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md:226]: Thanks for bringing this up! Yeah probably this is not really clear from the examples. This is explained more in the proposal part that type doesn't need to be there in the naming.  This is just a demonstration of how one of these types are used

File: teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md
Comment (ywluogg) [reply teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md:237]: That's a good point. I can definitely add it in the implementation, but let me bring this up for discussion. I will first implement the basic feature

**Pull request merged**
